Firebase installation

ruimagalhaes opened the pull request.

ruimagalhaes committed:
firebase module installation

ruimagalhaes committed:
firebase module added

Firebase module that will provide the context to be used on other components

ruimagalhaes committed:
firebase provider placed

placed the firebase provider arounf the digitransit-ui client

ruimagalhaes committed:
removed test firebase app details

tomasmcm commented on app/client.js:
```diff
@@ -192,36 +193,38 @@ const callback = () =>
+                      </ErrorBoundary>
+                    </ContextProvider>
+                  </ClientBreakpointProvider>
+                </FirebaseContext.Provider>
```
Would be nice if we did not break compatibility with upstream on the first day 😉
Suggestion:
- change `const content` to `let content` on line 195
- revert the content declaration to what was before
- add this after:
```
if(config.FIREBASE) {
  content = (
    <FirebaseContext.Provider value={new Firebase(config.FIREBASE)}>
      { content }
    </FirebaseContext.Provider>
  )
}
```

ruimagalhaes replied:
> Although not the primary goal, I think we can take into consideration if changes can be toggled with config vars. If it's simple to toggle, it's easier to add the toggle now than when we try to merge with upstream (either to get updates or to push our features).

@tomasmcm I'm afraid that I'm not getting exactly what you mean. Can you noobify that for me? :P

tomasmcm replied:
@ruimagalhaes If we can add changes without breaking the HSLdevcom/digitransit-ui default configs I think we should. That's why I've been adding and using config variables (in the config.porto.js file).
There are two advantages with this: first, it's easier to merge their master with ours to get updates, second, if we decide to submit a PR to their master we shouldn't need to change much (because our features will all be optional based on the loaded config).

tomasmcm reviewed (commented):
Although not the primary goal, I think we can take into consideration if changes can be toggled with config vars. If it's simple to toggle, it's easier to add the toggle now than when we try to merge with upstream (either to get updates or to push our features).

ruimagalhaes commented:
@tomasmcm thanks! will add the 3rd party auth methods and then apply those changes too

ruimagalhaes committed:
added google signin methods

ruimagalhaes committed:
HOC for wrapping components that need auth data

ruimagalhaes committed:
Auth button changed to use the auth HOC

ruimagalhaes commented:
@tomasmcm added the HOC for handling the auth state

tomasmcm commented:
@ruimagalhaes can you add the env vars needed here? https://github.com/whitesmith/digitransit-ui/wiki
thanks 😄

ruimagalhaes commented:
> @ruimagalhaes can you add the env vars needed here? https://github.com/whitesmith/digitransit-ui/wiki
> thanks 😄

@ruimagalhaes we shouldn't have those public I guess. I'll pass you the test data that I'm using

tomasmcm commented:
> @ruimagalhaes we shouldn't have those public I guess. I'll pass you the test data that I'm using

@ruimagalhaes yes, you can add the test data. at the bare minimum try to add the variable names that are required (that way we know if we are missing something)

tomasmcm commented on app/component/AuthButton.js:
```diff
@@ -1,39 +1,35 @@
-  } else {
-    return navAuthButton("signin", "sign-in", "Sign in", () => {
-      console.log("firebase.auth().signInWithRedirect") 
+class AuthButtonBase extends Component {
```
their pattern is to extend `React.Component`

tomasmcm committed:
fix warnings

isRequired is causing an error on AuthButton, since the firebase object
is a prop of AuthButton and not of withAuthentication
